core/signer/signer.go

```diff
@@ -64,9 +64,21 @@ func New(realm string,
 }
 
 // StartSigning advance epochs once per interval.
-func (s *Signer) StartSigning(ctx context.Context, interval time.Duration) {
-	for range time.NewTicker(interval).C {
-		if err := s.CreateEpoch(ctx); err != nil {
+func (s *Signer) StartSigning(ctx context.Context, minInterval time.Duration, maxElapsed uint) {
+	// count elapsed minIntervals:
+	elapsed := uint(0)
+	for range time.NewTicker(minInterval).C {
+		elapsed++
+		var createNewEpoch bool
+		if elapsed >= maxElapsed {
+			createNewEpoch = true
+			elapsed = 0
```

Comment:
How about we use a time here for this instead of a counter, this way the unit of `maxElapsed` is not coupled with `minInterval`? What do you think?